pkg/types/vsphere/validation/platform.go

```diff
@@ -14,6 +16,10 @@ import (
 func ValidatePlatform(p *vsphere.Platform, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 	if len(p.VCenter) == 0 {
+		vCenter := p.VCenter
```

Comment:
why the initialization?

Reply:
also you are testing this in an if block that is only true when vCenter is zero... so i think this one needs to outside the if block.